hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java

```diff
@@ -371,27 +350,50 @@ public void deleteByControllerID(final String controllerID) {
     }
 
     @Override
-    public Page<Target> findByAssignedDistributionSet(final Pageable pageReq, final long distributionSetID) {
-        throwEntityNotFoundIfDsDoesNotExist(distributionSetID);
+    public Page<Target> findByAssignedDistributionSet(final Pageable pageReq, final long distributionSetId) {
+        throwEntityNotFoundIfDsDoesNotExist(distributionSetId);
 
-        return targetRepository.findByAssignedDistributionSetId(pageReq, distributionSetID);
+        return targetRepository.findByAssignedDistributionSetId(pageReq, distributionSetId);
     }
 
     @Override
-    public Page<Target> findByAssignedDistributionSetAndRsql(final Pageable pageReq, final long distributionSetID,
-            final String rsqlParam) {
-        throwEntityNotFoundIfDsDoesNotExist(distributionSetID);
+    public Page<Target> findByAssignedDistributionSetAndQuery(final Pageable pageable, final long distributionSetId,
+            final String query) {
+        throwEntityNotFoundIfDsDoesNotExist(distributionSetId);
+        // Maybe optimize in order to fetch only the IDs from the database:
```

Comment:
remove comment and apply optimization, for example, use native query in repository to select only ids

Reply:
Done - drawback of native (jpa) queries is, that they are not taken into account when a refactoring takes place.